connectors/src/connectors/zendesk/lib/zendesk_api.ts

```diff
@@ -94,6 +94,45 @@ async function handleZendeskRateLimit(response: Response): Promise<boolean> {
   return false;
 }
 
+/**
+ * Runs a GET request to the Zendesk API with a maximum number of retries before throwing.
+ */
+async function fetchWithRetries({
```

Comment:
@aubin-tchoi  dont we have one already ? (`withRetries` comes to mind)

Reply:
Yup 

You can keep that function though, just remove the retry logic and use
```import { withRetries } from "@connectors/lib/dust_front_api_helpers";```

Reply:
doesn't seem like it does exactly what I need 🤷 
In my case I don't know the `delayBetweenRetriesMs` beforehand and it is not the same for every retry since I wait for a duration specified in the response.

Reply:
Let's call it fetchWithRetriesAndZendeskRateLimit then?